### ISSUE TITLE
llistxattr breaks compilation on Mac OS X

### ISSUE THREAD

**Issue by rfjakob** [vgough/encfs#305]
The patch https://github.com/vgough/encfs/pull/287 was merged to fix the issue https://github.com/vgough/encfs/issues/247.

However, it breaks Mac OS X because that does not have `llistxattr` as reported in https://github.com/vgough/encfs/commit/62505fa240e8e1c64c21df99d2ab9967a2636bf7#commitcomment-21253470 .

What to do here?

**Comment by rfjakob** [comment]
@tsjk I guess we have to revert to the buggy `listxattr` on Mac OS X?

**Comment by benrubson** [comment]
What about this ?
Simply check if xattr function `llistxattr` exists in CMakeLists.txt.
If it does not exist, then :
`#define llistxattr listxattr`

**Comment by benrubson** [comment]
See (and test) #306 for a fix ?

**Comment by rfjakob** [comment]
@tsjk Can you

    git clone -b llistxattr https://github.com/benrubson/encfs.git

an see if that fixes the issue?

**Comment by tsjk** [comment]
@rfjakob : it doesn't;
```
:info:build [ 79%] Building CXX object CMakeFiles/encfs.dir/encfs/MACFileIO.cpp.o
:info:build /usr/bin/clang++   -DELPP_DISABLE_DEFAULT_CRASH_HANDLING -DELPP_NO_DEFAULT_LOG_FILE -DELPP_SYSLOG -DELPP_THREAD_SAFE -DENABLE_NLS -DFUSE_USE_VERSION=26 -DLOCALEDIR=\"/opt/local/share/locale\" -DPACKAGE=\"encfs\" -D_FILE_OFFSET_BITS=64 -I/opt/local/include/osxfuse -I/opt/local/include -I/opt/local/var/macports/build/_usr_local_macports_ports_fuse_encfs-devel/encfs-devel/work/encfs-62505fa240e8e1c64c21df99d2ab9967a2636bf7/internal/tinyxml2-3.0.0 -I/opt/local/var/macports/build/_usr_local_macports_ports_fuse_encfs-devel/encfs-devel/work/build -I/opt/local/var/macports/build/_usr_local_macports_ports_fuse_encfs-devel/encfs-devel/work/encfs-62505fa240e8e1c64c21df99d2ab9967a2636bf7  -pipe -Os -DNDEBUG -stdlib=libc++ -arch x86_64 -mmacosx-version-min=10.11   -std=gnu++11 -o CMakeFiles/encfs.dir/encfs/MACFileIO.cpp.o -c /opt/local/var/macports/build/_usr_local_macports_ports_fuse_encfs-devel/encfs-devel/work/encfs-62505fa240e8e1c64c21df99d2ab9967a2636bf7/encfs/MACFileIO.cpp
:info:build /opt/local/var/macports/build/_usr_local_macports_ports_fuse_encfs-devel/encfs-devel/work/encfs-62505fa240e8e1c64c21df99d2ab9967a2636bf7/encfs/encfs.cpp:715:15: error: no member named 'llistxattr' in the global namespace; did you mean 'listxattr'?
:info:build   int res = ::llistxattr(cyName.c_str(), list, size, options);
:info:build             ~~^~~~~~~~~~
:info:build               listxattr
:info:build /usr/include/sys/xattr.h:73:9: note: 'listxattr' declared here
:info:build ssize_t listxattr(const char *path, char *namebuff, size_t size, int options);
:info:build         ^
```

**Comment by tsjk** [comment]
At least it doesn't build.

**Comment by benrubson** [comment]
Let me then try to correct the PR, I'll come back shortly !

**Comment by benrubson** [comment]
@tsjk could you give the whole output of :
```
git clone -b llistxattr https://github.com/benrubson/encfs.git
cd encfs
rm -rf build
mkdir build
cd build
cmake --version
cmake ..
grep XATTR_LLIST *
```

Thx !

**Comment by tsjk** [comment]
I cannot do this today because my mac system is shut down on Wednesdays (due to house cleaning :) ), but I'll tend to helping out with this again tomorrow.

**Comment by tsjk** [comment]
I think I missed that I should use a specific branch. 544275f24b5b1ae20f08ed1f00d64269665dee74 built successfully.

**Comment by benrubson** [comment]
Good news !
@rfjakob should then be able to merge 👍

**Comment by rfjakob** [comment]
Merged #307
